contracts/distribution.js

```diff
@@ -0,0 +1,287 @@
+    && api.assert(isSignedWithActiveKey === true, 'you must use a transaction signed with your active key')
+    && api.assert(api.sender === api.owner || api.sender === dist.creator, 'must be contract owner or creator')) {
+    if (await processBatch(dist, symbol, true)) {
+      api.emit('flush', { memo: `${symbol} payout distributed` });
+    }
+  }
+};
+
+actions.update = async (payload) => {
+  const {
+    id, tokenMinPayout, tokenRecipients, isSignedWithActiveKey,
+  } = payload;
+
+  // get contract params
+  const params = await api.db.findOne('params', {});
+  const { distUpdateFee } = params;
+
+  // eslint-disable-next-line no-template-curly-in-string
+  const utilityTokenBalance = await api.db.findOneInTable('tokens', 'balances', { account: api.sender, symbol: "'${CONSTANTS.UTILITY_TOKEN_SYMBOL}$'" });
+
+  const authorizedCreation = api.BigNumber(distUpdateFee).lte(0) || api.sender === api.owner
+    ? true
+    : utilityTokenBalance && api.BigNumber(utilityTokenBalance.balance).gte(distUpdateFee);
+
+  if (api.assert(authorizedCreation, 'you must have enough tokens to cover the update fee')
+    && api.assert(isSignedWithActiveKey === true, 'you must use a transaction signed with your active key')) {
+    const exDist = await api.db.findOne('batches', { _id: id });
+    if (api.assert(exDist, 'distribution not found') && await validateMinPayout(tokenMinPayout)
+      && await validateRecipients(tokenRecipients)) {
+      exDist.tokenMinPayout = tokenMinPayout;
+      exDist.tokenRecipients = tokenRecipients;
+      await api.db.update('batches', exDist);
+
+      // burn the token creation fees
+      if (api.sender !== api.owner && api.BigNumber(distUpdateFee).gt(0)) {
+        await api.executeSmartContract('tokens', 'transfer', {
+          // eslint-disable-next-line no-template-curly-in-string
+          to: 'null', symbol: "'${CONSTANTS.UTILITY_TOKEN_SYMBOL}$'", quantity: distUpdateFee, isSignedWithActiveKey,
+        });
+      }
+      api.emit('update', { id: exDist._id });
+    }
+  }
+};
+
+actions.setActive = async (payload) => {
+  const {
+    id,
+    active,
+    isSignedWithActiveKey,
+  } = payload;
+
+  if (!api.assert(isSignedWithActiveKey === true, 'you must use a custom_json signed with your active key')) {
+    return;
+  }
+  const dist = await api.db.findOne('batches', { _id: id });
+  if (api.assert(dist, 'distribution id not found')
+    && api.assert(dist.creator === api.sender || api.owner === api.sender, 'you must be the creator of this distribution')) {
+    dist.active = !!active;
+    await api.db.update('batches', dist);
+  }
+};
+
+actions.deposit = async (payload) => {
+  const {
+    id, symbol, quantity, isSignedWithActiveKey,
+  } = payload;
+
+  if (!api.assert(isSignedWithActiveKey === true, 'you must use a custom_json signed with your active key')
+    && !api.assert(quantity && api.BigNumber(quantity).dp() <= 3 && api.BigNumber(quantity).gt(0), 'invalid quantity')) {
```

Comment:
this should be OR ing the asserts?

Reply:
should have a test to cover these asserts

Reply:
Nice catch and added to tests as well.